File: apps/api/src/Eventuras.Domain/PaymentMethod.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Eventuras.Domain;

public class PaymentMethod
{
    public enum PaymentProvider
    {
        EmailInvoice = 1,
        PowerOfficeEmailInvoice = 2,
        PowerOfficeEHFInvoice = 3,
        StripeInvoice = 4,
        StripeDirect = 5,
        VippsInvoice = 6,
        VippsDirect = 7
    }

    public enum PaymentProviderType
    {
        Direct,
        Invoice
    }

    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.None)]
    public PaymentProvider Provider { get; set; }

    public PaymentProviderType Type { get; set; }

    public string Name { get; set; }
    public bool Active { get; set; } = false;
    public bool AdminOnly { get; set; } = false;
    public bool IsDefault { get; set; } = false;
}

File: apps/web/src/lib/paymentMethod.ts
import type { PaymentProvider } from '@/lib/eventuras-sdk';

/**
 * Display labels for PaymentProvider values. Hardcoded until translations
 * are added.
 */
export const paymentMethodLabels: { value: PaymentProvider; label: string }[] = [
  { value: 'EmailInvoice', label: 'Email invoice' },
  { value: 'PowerOfficeEmailInvoice', label: 'PowerOffice email invoice' },
  { value: 'PowerOfficeEHFInvoice', label: 'PowerOffice EHF invoice' },
  { value: 'StripeInvoice', label: 'Stripe invoice' },
  { value: 'StripeDirect', label: 'Stripe (direct)' },
  { value: 'VippsInvoice', label: 'Vipps invoice' },
  { value: 'VippsDirect', label: 'Vipps (direct)' },
];

export const getPaymentMethodLabel = (method?: PaymentProvider | null): string => {
  if (!method) return '';
  return paymentMethodLabels.find(x => x.value === method)?.label ?? method;
};
